src/Tribe/Ticket_Object.php

```diff
@@ -545,18 +547,28 @@ public function is_in_stock() {
 
 			/** @var Tribe__Cache $cache */
 			$cache = tribe( 'cache' );
-			$key   = __METHOD__ . '-' . $this->ID;
+			$cache_key   = __METHOD__ . '-' . $this->ID;
+			$is_ticket_cache_enabled = $this->is_ticket_cache_enabled();
+
+			if ( $is_ticket_cache_enabled ) {
+				$cached = $cache->get( $cache_key, Cache::TRIGGER_SAVE_POST, null );
+
+				if ( $cached !== null && is_string( $cached ) && in_array( $cached, [ 'yes', 'no' ], true ) ) {
+					$is_in_stock = tribe_is_truthy( $cached );
+					$cache->set( $cache_key, $is_in_stock ? 'yes' : 'no', Cache::TRIGGER_SAVE_POST );
```

Comment:
The trigger should be the 4th argument, aren't we missing the expiration arg here?

Reply:
Good catch! Thank you; you're right, the expiration duration is missing.